solr/solr-ref-guide/src/neural-search.adoc

```diff
@@ -0,0 +1,324 @@
+
+Neural Search is an industry derivation from the academic field of https://www.microsoft.com/en-us/research/uploads/prod/2017/06/fntir2018-neuralir-mitra.pdf[Neural information Retrieval].
+
+== Neural Search Concepts
+
+=== Deep Learning
+
+More and more frequently, we hear about how Artificial Intelligence (AI) permeates every aspect of our lives.
+
+When we talk about AI we are referring to a superset of techniques that enable machines to learn and show intelligence like humans.
+
+Since computing power has strongly and steadily advanced in the recent past, AI has seen a resurgence lately and it is now used in many domains, including software engineering and Information Retrieval (the science that regulates Search Engines and similar systems).
+
+In particular the advent of https://en.wikipedia.org/wiki/Deep_learning[Deep Learning] introduced the use of deep neural networks to solve complex problems that could not be solved simply by an algorithm.
+
+Deep Learning can be used to produce a vector representation of both the query and the documents in a corpus of information.
+
+=== Dense Vector Representation 
+A Dense vector describes information as an array of elements, each of them explicitly defined.
+
+Various Deep Learning models such as https://en.wikipedia.org/wiki/BERT_(language_model)[BERT] are able to encode textual information as dense vectors, to be used for Dense Retrieval strategies.
+
+For additional information you can refer to this https://sease.io/2021/12/using-bert-to-improve-search-relevance.html[blog post].
+
+=== Dense Retrieval
+Given a dense vector `v` that models the information need, the easiest approach for providing dense vector retrieval would be to calculate the distance(euclidean, dot product, etc.) between `v` and each vector `d` that represents a document in the corpus of information.
+
+This approach is quite expensive, so many approximate strategies are currently under active research.
+
+The strategy implemented in Apache Lucene and used by Apache Solr is based on Navigable Small-world graph.
+
+It provides efficient approximate nearest neighbor search for high dimensional vectors.
+
+See https://doi.org/10.1016/j.is.2013.10.006[Approximate nearest neighbor algorithm based on navigable small world graphs [2014]] and https://arxiv.org/abs/1603.09320[this paper [2018]] for details.
+
+
+== Index Time
+This is the list of Apache Solr field types designed to support Neural Search:
+
+=== DenseVectorField
+The Dense Vector field gives the possibility of indexing and searching dense vectors of float elements.
+
+e.g.
+
+`[1.0, 2.5, 3.7, 4.1]` (array of float elements)
+
+Here's how `DenseVectorField` should be configured in the schema:
+
+[source,xml]
+<fieldType name="knn_vector" class="solr.DenseVectorField" vectorDimension="4" similarityFunction="cosine"/>
+<field name="vector" type="knn_vector" indexed="true" stored="true"/>
+
+`vectorDimension`::
++
+[%autowidth,frame=none]
+|===
+|Mandatory
+|===
++
```

Comment:
The structure for this is always to include the default cell even if there is no default ("Default: none", and the word "Mandatory" should be "Required" in bold. See cluster-plugins.adoc for an example:

```
[%autowidth,frame=none]
|===
s|Required |Default: none
|===
```

Reply:
Done!